src/matches.rs

```diff
@@ -59,6 +62,16 @@ impl LateLintPass for MatchPass {
                          expression to match: `match *{} {{ ...`", snippet(cx, ex.span, "..")));
                 }
             }
+
+            // check preconditions for MATCH_BOOL
+            // type of expression == bool
+            if is_bool_expr(cx, ex) {
+                if in_external_macro(cx, expr.span) { return; }
```

Comment:
There's another `in_external_macro(..)` check higher up in the code. Since the tests aren't order dependent, as you already noticed, we can reorder them to get rid of the duplication.